addons/stock/models/product.py

```diff
@@ -783,6 +787,7 @@ def action_open_product_lot(self):
         if self.product_variant_count == 1:
             action['context'].update({
                 'default_product_id': self.product_variant_id.id,
+                'default_company_id': (self.company_id or self.env.company).id,
```

Comment:
Oy @Feyensv !
I think it's better to set the `default_company_id` directly at the line 786, when we define the context, instead of doing it in the condition, and thus set the `company_id`by default in all case, isn't ?

Reply:
Indeed :+1: